picard/formats/id3.py

```diff
@@ -494,6 +494,19 @@ def _get_tags(self, filename):
         except mutagen.id3.ID3NoHeaderError:
             return compatid3.CompatID3()
 
+    def _load_preserved_config(self, metadata):
+        # Adding additional config to metadata to prevent unecessary saving
+        metadata['~config:rating_steps'] = config.setting['rating_steps']
+        metadata['~config:rating_user_email'] = config.setting['rating_user_email']
```

Comment:
1. I still believe that this should NOT be stored on a file basis - it is a sledgehammer to crack a nut. Saving a file unnecessarily is not the end of the world. Saving thousands of files unnecessarily is - because of the performance issues.

2. I believe we need a constant which defines the list of settings that potentially affect tags not based on `metadata` and then your code should iterate using this list. Then if we need to add new settings, we just add them to the list.